Dockerfile

```diff
@@ -36,7 +36,7 @@ RUN cargo build --release --target $(bash target_arch.sh) --bin qdrant
 
 RUN mv target/$(bash target_arch.sh)/release/qdrant /qdrant/qdrant
 
-FROM debian:11-slim
+FROM --platform=${BUILDPLATFORM:-linux/amd64} debian:11-slim
```

Comment:
@timvisee can you elaborate on this, please?

As far as I understand, this will cause any cross-compilation builds (e.g., `x86 host + arm target` and `arm host + x86 target`) to use ___host___ architecture image as a base and copy ___target___ architecture binary into it.

I guess this _may_ work (as we only install `ca-certificates` and `tzdata` packages and Qdrant binary itself should be statically compiled), but it _feels_ wrong (and, I think, may break at some point and start producing broken images).

Reply:
You're right.

The problem is that cross compiling without this change doesn't work either. This is rather a quick fix for that.

Right now, we want to install `ca-certificates` and `tzdata`. For that we require `apt-get` (a shell, and more). These cannot be executed because the binaries in this rootfs obviously have the wrong architecture.

Here are the two problematic instructions:

```Dockerfile
# -- snip --
RUN apt-get update \
    && apt-get install -y ca-certificates tzdata \
    && rm -rf /var/lib/apt/lists/*
RUN mkdir -p ${APP}
# -- snip --
```

Since no binaries (other than `qdrant`) are used at runtime, and these two packages don't install any additional binaries, I figured this would be good enough for now.

I believe there are two options to improve on this:
1. Use the target architecture in the final stage, and copy in `ca-certificates` and `tzdata`.
2. Base the final stage upon `scratch` and copy in all required runtime data.

The first should be quite easy. We leave it as is, and have to pull the two packages from a previous stage that does run on the host machine.

What would be even better is the second option. We built from `scratch` instead and copy in all required runtime data. This significantly decreases attack vectors, because basic tools such as a shell are not available making breaking out of the container much harder if a malicious user somehow gets access to the environment.

Sorry for the long message. What do you think? Can we leave this as is for now and implement either of the two options later, or should this be worked on now?

Reply:
> As far as I understand, Docker would run non-native build steps (e.g., if you are building arm64 Docker image on an amd64 host, or vise versa) though virtualization/emulation.

That's what I thought too, but this failed somehow. Maybe I'm having issues with my Docker installation.